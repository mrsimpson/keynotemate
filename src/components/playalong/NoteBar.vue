<template>
  <div 
    class="note-bar"
    :class="{ 'is-rest': note.isRest }"
    :style="style"
  >
    <div class="note-content">
      <span v-if="!note.isRest" class="note-label">
        {{ formatNoteDisplay }}
      </span>
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import type { ParsedNote } from '../../types/musicxml';
import { useNoteVisualizer } from '../../composables/useNoteVisualizer';
import { formatNoteForDisplay } from '../../utils/noteMapping';

const props = defineProps<{
  note: ParsedNote;
}>();

const { getNoteStyle } = useNoteVisualizer();
const style = getNoteStyle(props.note);

const formatNoteDisplay = computed(() => {
  if (props.note.isRest) return '';
  return formatNoteForDisplay(props.note.pitch, props.note.octave);
});</script>

<style scoped>
.note-bar {
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: var(--radius-md);
  transition: all var(--transition-normal);
  box-shadow: var(--shadow-sm);
  box-sizing: border-box;
  padding: 0;
  overflow: hidden;
}

.note-bar:hover {
  transform: translateY(-2px);
  box-shadow: var(--shadow-md);
}

.note-bar.is-rest {
  background-color: #f8f9fa;
  border: 1px dashed #ddd;
}

.note-content {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.note-label {
  font-weight: 600;
  letter-spacing: 0.5px;
  white-space: nowrap;
  padding: 0 var(--spacing-sm);
  font-size: 0.875rem;
}
</style>
