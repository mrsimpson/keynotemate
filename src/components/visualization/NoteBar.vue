<template>
  <div
    class="note-bar"
    :style="{
      backgroundColor: color,
      width: `${duration * 50}px`,
    }"
  >
    <span class="note-label">{{ note }}</span>
  </div>
</template>

<script setup lang="ts">
defineProps<{
  note: string;
  color: string;
  duration: number;
}>();
</script>

<style scoped>
.note-bar {
  height: 40px;
  min-width: 50px;
  margin: 4px 0;
  border-radius: 4px;
  display: flex;
  align-items: center;
  padding: 0 8px;
  transition: width 0.3s ease;
}

.note-label {
  color: white;
  text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.3);
  font-weight: bold;
}
</style>
