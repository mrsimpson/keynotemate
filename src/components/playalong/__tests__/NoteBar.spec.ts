import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import type { ParsedNote } from '@/types/musicxml.ts'
import NoteBar from '@/components/playalong/NoteBar.vue'

describe('NoteBar Component', () => {
  const mockNote: ParsedNote = {
    pitch: 'C',
    duration: {
      divisions: 4,
      type: 'quarter',
      relativeLength: 1,
    },
    octave: 4,
    isRest: false,
  }

  const mockRestNote: ParsedNote = {
    pitch: 'rest',
    duration: {
      divisions: 4,
      type: 'quarter',
      relativeLength: 1,
    },
    octave: 4,
    isRest: true,
  }

  it('renders note bar snapshot', () => {
    const wrapper = mount(NoteBar, {
      props: {
        note: mockNote,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders rest note bar snapshot', () => {
    const wrapper = mount(NoteBar, {
      props: {
        note: mockRestNote,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
