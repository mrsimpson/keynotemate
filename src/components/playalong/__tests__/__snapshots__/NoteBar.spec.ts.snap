// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoteBar Component > renders note bar snapshot 1`] = `
"<div data-v-a4d447ea="" class="note-bar" style="background-color: rgb(255, 0, 0); color: rgb(255, 255, 255); width: 60px; min-width: 40px;">
  <div data-v-a4d447ea="" class="note-content"><span data-v-a4d447ea="" class="note-label">C4</span></div>
</div>"
`;

exports[`NoteBar Component > renders rest note bar snapshot 1`] = `
"<div data-v-a4d447ea="" class="is-rest note-bar" style="width: 60px; min-width: 40px;">
  <div data-v-a4d447ea="" class="note-content">
    <!--v-if-->
  </div>
</div>"
`;
